site/gatsby-site/src/components/users/UserCreationDateCell.js

```diff
@@ -0,0 +1,22 @@
+import React from 'react';
+import { format } from 'date-fns';
+import { FIND_USER } from '../../graphql/users';
+import { useQuery } from '@apollo/client/react';
+import { Spinner } from 'flowbite-react';
+
+const UserCreationDateCell = ({ userId }) => {
+  const { data, loading } = useQuery(FIND_USER, {
```

Comment:
IIUC this will create separate parallel requests for the same user object for each of the user's creation date, email, auth date, etc fields. 

Is there a way to only request the user once and render the row information only on return?

I'm concerned about creating a large number of requests overall to Atlas, too, in case the traffic is "bursty" and gets limited, or for cause of billing.

What do you think? Can you measure/count the outgoing requests?

Reply:
(If it improves our performance today and doesn't cause issues, I would be happy to leave this as a future optimization. We shouldn't block fixing the test on this further optimization.)

Reply:
The `useQuery` hook has an [in-memory cache by default](https://www.apollographql.com/docs/react/data/queries/#cache-first) (`cache-first`). So we only call the Atlas graphql API once for each user.
This also improves the pagination performance since it won't refetch from the server if you go back to a previous page.

<img width="1427" alt="Screenshot 2023-09-08 at 11 59 02" src="https://github.com/responsible-ai-collaborative/aiid/assets/6564809/c0e1080a-8a5d-4579-b2fa-f042485a0dcf">